src/formElementPlugins/number/src/ApplicationView.tsx

```diff
@@ -74,26 +77,45 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
 
   function handleChange(e: any) {
     const text = e.target.value
-    const [number, _] = parseInput(text, numberFormatter, simple, prefix, suffix)
+    const { number } = parseInput(text, numberFormatter, simple, prefix, suffix, suffixPlural)
     setInternalValidation(customValidate(number, type, minValue, maxValue))
     onUpdate(text)
     setTextValue(text)
   }
 
   function handleLoseFocus() {
     if (!textValue) return
-    const [number, text] = parseInput(textValue, numberFormatter, simple, prefix, suffix)
+    const { number, formattedNumber, fullText } = parseInput(
+      textValue,
+      numberFormatter,
+      simple,
+      prefix,
+      suffix,
+      suffixPlural
+    )
     const validation = customValidate(number, type, minValue, maxValue)
     setInternalValidation(validation)
-    if (validation.isValid) onSave({ text, number, type, currency, locale })
+    if (validation.isValid)
+      onSave({
+        text: fullText,
+        number,
+        type,
+        currency,
+        locale,
+        prefix,
+        suffix,
+        suffixPlural,
```

Comment:
May as well store these values with the response, so the strings can be rebuilt from parts if required.